### PR TITLE
fix(backend): handle empty jina search results

### PR DESCRIPTION
**Pull request by RykerFeng**
### Why / What / How

Why: `SearchTheWebBlock` currently turns Jina's `422 No search results available` response into a block execution failure, which matches open issue #11632.
What: Treat that specific `422` response as an empty search result instead of an unknown block error, and add regression tests for the empty-result path plus a non-422 error path.
How: Detect the Jina no-results error message in `SearchTheWebBlock`, return `results=""` early for that case, and keep raising `BlockExecutionError` for other client and unexpected failures.

### Changes 🏗️

- Handle Jina `422 No search results available` responses in `SearchTheWebBlock` without failing the block
- Preserve existing failure behavior for other client errors
- Add regression coverage for both the empty-results path and the non-422 error path in the Jina block tests
- No configuration or environment changes

#### For code changes:
- [ ] I have clearly listed my changes in the PR description
- [ ] I have made a test plan
- [ ] I have tested my changes according to the test plan:
  - [ ] Run `PYTHONPATH=. ./.venv/bin/pytest test/blocks/test_jina_extract_website.py`
  - [ ] Run `PATH="$PWD/.venv/bin:$PATH" ./.venv/bin/poetry run lint --skip-pyright`

<details>
  <summary>Example test plan</summary>
  
  - [ ] Create from scratch and execute an agent with at least 3 blocks
  - [ ] Import an agent from file upload, and confirm it executes correctly
  - [ ] Upload agent to marketplace
  - [ ] Import an agent from marketplace and confirm it executes correctly
  - [ ] Edit an agent from monitor, and confirm it executes correctly
</details>

#### For configuration changes:

- [ ] `.env.default` is updated or already compatible with my changes
- [ ] `docker-compose.yml` is updated or already compatible with my changes
- [ ] I have included a list of my configuration changes in the PR description (under **Changes**)

<details>
  <summary>Examples of configuration changes</summary>

  - Changing ports
  - Adding new services that need to communicate with each other
  - Secrets or environment variable changes
  - New or infrastructure changes such as databases
</details>

### CODE REVIEW

File: autogpt_platform/backend/test/blocks/test_jina_extract_website.py [hunk 1]
```diff
@@ -47,7 +48,7 @@ async def test_extract_website_content_handles_http_error(monkeypatch):
         raw_content=False,
     )
 
-    async def fake_get_request(url, json=False, headers=None):
+    async def fake_get_request(_url, json=False, headers=None):
```
Comment (majdyz) [inline autogpt_platform/backend/test/blocks/test_jina_extract_website.py]: 🤖 🔵 **Nit**: The rename `url` → `_url` in `fake_get_request` on line 51 is out of scope — this PR fixes empty-search-results for `SearchTheWebBlock`, not the existing extract-website test.

1. PR scope is "handle empty jina search results" (Why/What/How all mention search only).
2. This file already had `url` and the underscore rename is just a style preference for unused params.
3. Mixing unrelated edits into a fix PR makes the diff harder to review and can mask real changes.

Would you mind reverting the `url` → `_url` change on line 51 (the original extract-website test) and keeping the underscore convention only for the two NEW search tests you are adding? Welcome to AutoGPT, and thanks for fixing #11632!

File: autogpt_platform/backend/test/blocks/test_jina_extract_website.py [hunk 2]
```diff
@@ -64,3 +65,51 @@ async def fake_get_request(url, json=False, headers=None):
...
+            422,
+        )
+
+    monkeypatch.setattr(block, "get_request", fake_get_request)
+
+    results = [
+        output
+        async for output in block.run(
+            input_data=input_data, credentials=TEST_CREDENTIALS
+        )
```
Comment (majdyz) [inline autogpt_platform/backend/test/blocks/test_jina_extract_website.py:96]: 🤖 🟠 **Should Fix**: The new test only asserts on the success path of the no-results branch, not on cost-stats behavior. Since you skip `merge_stats(...)` on early return (correctly — Jina 422 is unbilled), a future refactor could silently start charging users for empty searches without breaking tests.

1. `test_search_the_web_returns_empty_results_for_no_matches` only checks `results == [("results", "")]` (line 90).
2. The block records `provider_cost=0.01` via `merge_stats` on the success path; the early-return path must NOT.
3. If someone later moves `merge_stats` above the `try`, users get charged for the no-results path and no test catches it.

Could you also assert that no provider-cost stats were recorded on the no-results path (e.g. inspect `block.execution_stats` or `block._stats`), so the "free on 422" invariant is locked in?

**Comment by majdyz** [comment]
Hey @18062706139fcz — thanks for the contribution! The PR is otherwise in good shape but the CLA check is currently failing. Could you sign the CLA when you have a moment? It's a one-time step (the bot link should be in the failing check above) and unblocks the merge. Appreciated!

**Comment by 18062706139fcz** [comment]
> Hey @18062706139fcz — thanks for the contribution! The PR is otherwise in good shape but the CLA check is currently failing. Could you sign the CLA when you have a moment? It's a one-time step (the bot link should be in the failing check above) and unblocks the merge. Appreciated!

@majdyz  Thanks for the reminder! The CLA check didn’t go through earlier due to an account issue when I clicked the link, but it has been fixed now and the CLA should be signed successfully.

Could you please check again and see if the PR can be merged now? Thanks!

File: autogpt_platform/backend/backend/blocks/jina/search.py [hunk 3]
```diff
@@ -63,6 +115,16 @@ async def run(
             results = await self.get_request(
                 jina_search_url, headers=headers, json=False
             )
+        except HTTPClientError as e:
+            if _is_no_results_error(e):
+                # No merge_stats here: Jina does not bill the 422 no-results path.
+                yield "results", NO_SEARCH_RESULTS_OUTPUT
```
Comment (18062706139fcz) [inline autogpt_platform/backend/backend/blocks/jina/search.py:121]: 🤖 🔴 **Blocker**: This no longer models an empty-result search; it injects a synthetic sentence into the `results` pin. That changes downstream behavior in exactly the way a fallback/cover-up would: truthiness checks now see "results present", and LLM / templating flows consume fabricated content instead of a genuinely empty payload. The PR description says this path should be treated as an empty search result, so this should stay `""` or expose a separate explicit signal like `has_results`/`error`.

File: docs/integrations/block-integrations/misc.md [hunk 4]
```diff
@@ -58,7 +58,7 @@ Tool and block identifiers provided in `tools` and `blocks` are validated at run
...
... \| "bash_exec" \| "browser_act" \| "browser_navigate" \| "browser_screenshot" \| "connect_integration" \| "continue_run_block" \| "create_agent" \| "create_feature_request" \| "create_folder" \| "customize_agent" \| "decompose_goal" \| "delete_folder" \| "delete_schedule" \| "delete_skill" \| "delete_workspace_file" \| "edit_agent" \| "find_agent" \| "find_block" \| "find_library_agent" \| "fix_agent_graph" \| "get_agent_building_guide" \| "get_doc_page" \| "get_mcp_guide" \| "get_platform_info" \| "get_sub_session_result" \| "list_agent_triggers" \| "list_folders" \| "list_schedules" \| "list_skills" \| "list_workspace_files" \| "memory_forget_confirm" \| "memory_forget_search" \| "memory_search" \| "memory_store" \| "move_agents_to_folder" \| "move_folder" \| "read_skill" \| "read_workspace_file" \| "run_agent" \| "run_block" \| "run_mcp_tool" \| "run_sub_session" \| "schedule_followup" \| "search_docs" \| "search_feature_requests" \| "store_skill" \| "update_folder" \| "validate_agent_graph" \| "view_agent_output" \| "web_fetch" \| "web_search" \| "write_workspace_file" \| "Agent" \| "Edit" \| "Glob" \| "Grep" \| "Read" \| "Task" \| "TodoWrite" \| "WebSearch" \| "Write"] | No |
```
Comment (18062706139fcz) [inline docs/integrations/block-integrations/misc.md:61]: 🤖 🔵 **Nit**: This documentation change is unrelated to the Jina empty-results fix. Adding `decompose_goal` to the tool list broadens the PR scope and makes the review noisier without helping validate the bug fix. Please drop it into a separate docs/tooling PR unless this block change truly depends on it.
